lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -109,4 +121,31 @@ export abstract class ClientApplication {
             },
         };
     }
+
+    /**
+     * Create authority instance. If authority not passed in request, default to authority set on the application
+     * object. If no authority set in application object, then default to common authority.
+     * @param authorityString
+     */
+    private async createAuthority(authorityString?: string): Promise<Authority> {
```

Comment:
Couldn't you accomplish the same thing by exposing this as a public API on the base client? Then this could be used in any application that uses msal-common, and doesn't force msal-browser and msal-node to implement the same function.

Reply:
Ideally yes but we need to set _authority on the ClientApplication object, which we can't do from   common.

Reply:
the public API can return an authority object that you set in ClientApplication. basically before you pass the config object you can call `AuthorityFactory.createAuthority()` with the function signature here and then use that authority for all calls. This way I don't need to need to write duplicate code in the browser, I can simply call the same API. See [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/1584/files/06ce0a617f343af9399e9b63bf5b8f7533116bbe#diff-3e92fba27aeb0efe69aaa786b5f9649b)

Reply:
Just ported the changes for browser, and this code is actually not needed there. Can you take a look and see if the changes make sense? Essentially since browser only uses SPAClient, this is not applicable, and thus there is only one default authority, this is not applicable.